tools/build_release.py

```diff
@@ -21,109 +20,137 @@
-
-    parser.add_option("", "--report-build", dest="report_build_file_name", help="Output the build results to an junit xml file")
-
-    parser.add_option("", "--build-tests", dest="build_tests", help="Build all tests in the given directories (relative to /libraries/tests)")
-
-
+    parser.add_option(
+        "-p", "--platforms",
+        dest="platforms",
+        default="",
+        help="Build only for the platform namesseparated by comma"
+    )
+    parser.add_option(
+        "-L", "--list-config",
+        action="store_true",
+        dest="list_config",
+        default=False,
+        help="List the platforms and toolchains in the release in JSON"
+    )
+    parser.add_option(
+        "", "--report-build",
+        dest="report_build_file_name",
+        help="Output the build results to an junit xml file"
+    )
+    parser.add_option(
+        "", "--build-tests",
+        dest="build_tests",
+        help="Build all tests in the given directories"
+        " (relative to /libraries/tests)"
+    )
     options, args = parser.parse_args()
-
-
-
     if options.list_config:
-        print json.dumps(OFFICIAL_MBED_LIBRARY_BUILD, indent=4)
+        print(json.dumps(OFFICIAL_MBED_LIBRARY_BUILD, indent=4))
         sys.exit()
-
     start = time()
     build_report = {}
     build_properties = {}
-
     platforms = None
     if options.platforms != "":
         platforms = set(options.platforms.split(","))
-
     status = True
-
     if options.build_tests:
         # Get all paths
         directories = options.build_tests.split(',')
         for i in range(len(directories)):
             directories[i] = normpath(join(TEST_DIR, directories[i]))
-
         test_names = []
-
-        for test_id in TEST_MAP.keys():
+        for test_id in list(TEST_MAP.keys()):
             # Prevents tests with multiple source dirs from being checked
-            if isinstance( TEST_MAP[test_id].source_dir, basestring):
+            if isinstance(TEST_MAP[test_id].source_dir, str):
                 test_path = normpath(TEST_MAP[test_id].source_dir)
                 for directory in directories:
                     if directory in test_path:
                         test_names.append(test_id)
-
         mut_counter = 1
         mut = {}
         test_spec = {
             "targets": {}
         }
 
         if options.toolchains:
-            print "Only building using the following toolchains: %s" % (options.toolchains)
+            print("Only building using the following toolchains: {}".format(
+                options.toolchains
+            ))
 
         for target_name, toolchain_list in OFFICIAL_MBED_LIBRARY_BUILD:
             toolchains = None
-            if platforms is not None and not target_name in platforms:
-                print("Excluding %s from release" % target_name)
+            if platforms is not None and target_name not in platforms:
+                print(("Excluding %s from release" % target_name))
```

Comment:
Probably don't need double parenthesis 😄. But moving it to the `"".format()` style would probably be good:
```suggestion
                print("Excluding {} from release".format(target_name))
```

Reply:
Also a 2to3 thing. I'll take your suggestion.